### PR TITLE
ci: Run the sanitizers in parallel

### PR DESCRIPTION
**Pull request by larseggert**
This speeds up the matrix.

Also grab MSRV from `Cargo.toml` instead of hardcoding it.

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -25,34 +25,77 @@ concurrency:
...
     steps:
       - uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 # v4.2.0
 
       - uses: ./.github/actions/rust
         with:
-          version: ${{ matrix.rust-toolchain }}
-          components: ${{ matrix.rust-toolchain == 'stable' && 'llvm-tools-preview' || matrix.rust-toolchain == 'nightly' && 'rust-src' || '' }}
-          tools: ${{ matrix.rust-toolchain == 'stable' && 'cargo-llvm-cov, ' || '' }} cargo-nextest
+          version: ${{ matrix.toolchain }}
+          components: ${{ matrix.toolchain == 'stable' && 'llvm-tools-preview' || matrix.toolchain == 'nightly' && 'rust-src' || '' }}
```
Comment (mxinden) [inline .github/workflows/check.yml:97]:
```suggestion
          components: ${{ matrix.toolchain == 'stable' && 'llvm-tools-preview' || matrix.sanitizer != null && 'rust-src' || '' }}
```

Would be more accurate, right?

**Review by mxinden — APPROVED** [review]
Definitely in favor of the `toolchains` job.

In case the gain from parallelizing the sanitizers is worth the complexity, :+1:.

**Comment by larseggert** [comment]
Do you think I should just do a separate workflow for the sanitizer runs, like we do for clippy? May be more maintainable.

**Comment by larseggert** [comment]
@mxinden any ideas on what's causing the two failures?

**Review by mxinden — APPROVED** [review]

**Pull request merged**